Reduce symbol search

smuzaffar opened the pull request.

vgvassilev committed:
Revert "[core] Materialize symbols for TError variables"

This reverts commit 329fb5ae0c5717aec95fe50ec276f07450ac7f11.

smuzaffar commented:
please test for CMSSW_14_0_ROOT6_X

cmsbuild commented:
A new Pull Request was created by  @smuzaffar (Malik Shahzad Muzaffar) for branch cms/master/329fb5ae0c.

@aandvalenzuela, @iarspider, @smuzaffar can you please review it and eventually sign? Thanks.
@sextonkennedy, @rappoccio, @antoniovilela you are the release manager for this.
cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

cmsbuild commented:
cms-bot internal usage<!-- bot cache: {"commits":{"55145f099aed8f511450cdb0f9577e5113cd663c":{"files":["core/foundation/inc/TError.h","core/foundation/src/RLogger.cxx","core/foundation/src/TError.cxx"],"squashed":false,"time":1704961803},"566e73c9c1f2c54be1462481e6833b4c8ba70312":{"files":["core/metacling/src/TClingDataMemberInfo.cxx"],"squashed":false,"time":1704961730},"984d562aab4809643e9c7827500c858f8b2c9e51":{"files":["core/metacling/src/TCling.cxx","core/metacling/src/TClingCallbacks.cxx","core/metacling/src/TClingCallbacks.h","core/metacling/test/TClingDataMemberInfoTests.cxx"],"squashed":false,"time":1704961643},"c22be553e76a74a14b2b5d4f1e93b3279f61416c":{"files":["core/metacling/src/TClingDataMemberInfo.cxx"],"squashed":false,"time":1704961685}},"emoji":{},"last_seen_sha":"55145f099aed8f511450cdb0f9577e5113cd663c","signatures":{}} -->

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36678/summary.html
**COMMIT**: e2af2f9d42fe3860f3e5b453f5bbe65621ffe75b
**CMSSW**: CMSSW_14_0_ROOT6_X_2023-12-21-2300/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/root/193/36678/install.sh` to create a dev area with all the needed externals and cmssw changes.



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - @cms-sw cms-sw/cmsdist#8864
 - @cms-sw cms-sw/cmsdist#8913
 - @cms-sw cms-sw/cmsdist#8910
 - @cms-sw cms-sw/cmsdist#8911
 - @cms-sw cms-sw/cmsdist#8908
 - @jmduarte cms-sw/cmsdist#8906

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36678/git-recent-commits.json
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36678/git-merge-result


## Comparison Summary



**Summary**:
 - You potentially [removed 3 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_0_X_2023-12-21-2300+78f86a/60434/validateJR/logRootQA-events.log) from the logs
 - [Reco comparison results: 3879 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_0_X_2023-12-21-2300+78f86a/60434/validateJR.html)
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3247277
 - DQMHistoTests: [Total failures: 8009](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_0_X_2023-12-21-2300+78f86a/60434/validateJR.html)
 - DQMHistoTests: Total nulls: 20
 - DQMHistoTests: Total successes: 3239226
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 200 log files, 161 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_0_X_2023-12-21-2300+78f86a/60434/triggerResults): found differences in 1 / 46 workflows

vgvassilev committed:
[core] Prefer variable initializer over symbol search

Symbol lookup is a quite expensive operation and might result in JIT
compilation and library loading.

Co-authored-by: Jonas Hahnfeld <jonas.hahnfeld@cern.ch>

vgvassilev committed:
[core] Reduce symbol search only to when autoloading is enabled.

The llvm9 JIT issued callbacks when a symbol was missing and we reacted on it
by loading the relevant library. In root-project/root@9b2041e3 we have kept the
logic but now the JIT started querying more often even for symbols which are
okay to be missing. In turn that leads to scanning all libraries causing
performance issues.

This patch tries to limit this functionality only in contexts where automatic
loading is allowed.

vgvassilev committed:
[core] Evaluate initializer of constant variables

Co-authored-by: Jonas Hahnfeld <jonas.hahnfeld@cern.ch>

cmsbuild commented:
Pull request #193 was updated.

smuzaffar commented:
please test for CMSSW_14_0_ROOT6_X

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36736/summary.html
**COMMIT**: 8f4c26d66188c599b17207cc3dc39ada0fee5a13
**CMSSW**: CMSSW_14_0_ROOT6_X_2024-01-07-2300/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/root/193/36736/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially [added 87 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_0_X_2024-01-07-2300+78f86a/60491/validateJR/logRootQA-events.log) to the logs
 - [Reco comparison results: 993 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_0_X_2024-01-07-2300+78f86a/60491/validateJR.html)
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3247277
 - DQMHistoTests: [Total failures: 6270](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_0_X_2024-01-07-2300+78f86a/60491/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3240985
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 200 log files, 161 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_0_X_2024-01-07-2300+78f86a/60491/triggerResults): no differences found

cmsbuild commented:
Pull request #193 was updated.

smuzaffar commented:
please test for CMSSW_14_0_ROOT6_X

smuzaffar commented:
please test with cms-sw/cmsdist#8936 for CMSSW_14_0_ROOT6_X

cmsbuild commented:
-1

**Failed Tests**: UnitTests RelVals 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/summary.html
**COMMIT**: 55145f099aed8f511450cdb0f9577e5113cd663c
**CMSSW**: CMSSW_14_0_ROOT6_X_2024-01-14-2300/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/root/193/36826/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Unit Tests

I found 3 errors in the following unit tests:

<pre>---> test runtestRecoEgammaElectronIdentification had ERRORS
---> test testTauEmbeddingProducers had ERRORS
---> test testAlcaBeamMonitor had ERRORS
</pre>

## RelVals

- [136.7611](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/136.7611_RunJetHT2016EreMINIAOD)
```
----- Begin Fatal Exception 15-Jan-2024 21:58:19 CET-----------------------
An exception of category 'FileInPathError' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing module: class=ElectronMVAValueMapProducer label='electronMVAValueMapProducer'
Exception Message:
edm::FileInPath unable to find file RecoEgamma/ElectronIdentification/data/MVAWeightFiles/Winter22HZZV1/EB1_5.weights.xml.gz anywhere in the search path.
The search path is defined by: CMSSW_SEARCH_PATH
${CMSSW_SEARCH_PATH} is: /cvmfs/cms-ci.cern.ch/week0/cms-sw/root/193/36826/CMSSW_14_0_ROOT6_X_2024-01-14-2300/poison:/cvmfs/cms-ci.cern.ch/week0/cms-sw/root/193/36826/CMSSW_14_0_ROOT6_X_2024-01-14-2300/src:/cvmfs/cms-ci.cern.ch/week0/cms-sw/root/193/36826/CMSSW_14_0_ROOT6_X_2024-01-14-2300/external/el8_amd64_gcc12/data:/cvmfs/cms-ib.cern.ch/sw/x86_64/week0/el8_amd64_gcc12/cms/cmssw/CMSSW_14_0_ROOT6_X_2024-01-14-2300/src
Current directory is: /data/cmsbld/jenkins/workspace/ib-run-pr-relvals/matrix-results/136.7611_RunJetHT2016EreMINIAOD
----- End Fatal Exception -------------------------------------------------
```
- [136.8311](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/136.8311_RunJetHT2017FreMINIAOD)
```
----- Begin Fatal Exception 15-Jan-2024 21:58:19 CET-----------------------
An exception of category 'FileInPathError' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing module: class=ElectronMVAValueMapProducer label='electronMVAValueMapProducer'
Exception Message:
edm::FileInPath unable to find file RecoEgamma/ElectronIdentification/data/MVAWeightFiles/Winter22HZZV1/EB1_5.weights.xml.gz anywhere in the search path.
The search path is defined by: CMSSW_SEARCH_PATH
${CMSSW_SEARCH_PATH} is: /cvmfs/cms-ci.cern.ch/week0/cms-sw/root/193/36826/CMSSW_14_0_ROOT6_X_2024-01-14-2300/poison:/cvmfs/cms-ci.cern.ch/week0/cms-sw/root/193/36826/CMSSW_14_0_ROOT6_X_2024-01-14-2300/src:/cvmfs/cms-ci.cern.ch/week0/cms-sw/root/193/36826/CMSSW_14_0_ROOT6_X_2024-01-14-2300/external/el8_amd64_gcc12/data:/cvmfs/cms-ib.cern.ch/sw/x86_64/week0/el8_amd64_gcc12/cms/cmssw/CMSSW_14_0_ROOT6_X_2024-01-14-2300/src
Current directory is: /data/cmsbld/jenkins/workspace/ib-run-pr-relvals/matrix-results/136.8311_RunJetHT2017FreMINIAOD
----- End Fatal Exception -------------------------------------------------
```
- [136.88811](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/136.88811_RunJetHT2018DreMINIAODUL)
```
----- Begin Fatal Exception 15-Jan-2024 21:58:20 CET-----------------------
An exception of category 'FileInPathError' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing module: class=ElectronMVAValueMapProducer label='electronMVAValueMapProducer'
Exception Message:
edm::FileInPath unable to find file RecoEgamma/ElectronIdentification/data/MVAWeightFiles/Winter22HZZV1/EB1_5.weights.xml.gz anywhere in the search path.
The search path is defined by: CMSSW_SEARCH_PATH
${CMSSW_SEARCH_PATH} is: /cvmfs/cms-ci.cern.ch/week0/cms-sw/root/193/36826/CMSSW_14_0_ROOT6_X_2024-01-14-2300/poison:/cvmfs/cms-ci.cern.ch/week0/cms-sw/root/193/36826/CMSSW_14_0_ROOT6_X_2024-01-14-2300/src:/cvmfs/cms-ci.cern.ch/week0/cms-sw/root/193/36826/CMSSW_14_0_ROOT6_X_2024-01-14-2300/external/el8_amd64_gcc12/data:/cvmfs/cms-ib.cern.ch/sw/x86_64/week0/el8_amd64_gcc12/cms/cmssw/CMSSW_14_0_ROOT6_X_2024-01-14-2300/src
Current directory is: /data/cmsbld/jenkins/workspace/ib-run-pr-relvals/matrix-results/136.88811_RunJetHT2018DreMINIAODUL
----- End Fatal Exception -------------------------------------------------
```
<details>
<summary>Expand to see more relval errors ...</summary>

- [4.53](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/4.53_RunPhoton2012B)
- [135.4](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/135.4_ZEEFS_13)
- [140.023](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/140.023_RunZeroBias2022B)
- [139.001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/139.001_RunMinimumBias2021)
- [136.731](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/136.731_RunSinglePh2016B)
- [136.874](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/136.874_RunEGamma2018C)
- [136.793](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/136.793_RunDoubleEG2017C)
- [158.01](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/158.01_HydjetQB12ppRECOin2018reMINIAOD)
- [140.043](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/140.043_RunZeroBias2022C)
- [140.063](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/140.063_RunZeroBias2022D)
- [9.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/9.0_Higgs200ChargedTaus)
- [25.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/25.0_TTbar)
- [1306.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/1306.0_SingleMuPt1_UP15)
- [1330.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/1330.0_ZMM_13)
- [312.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/312.0_Pyquen_ZeemumuJets_pt10_2760GeV_2022)
- [141.042](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/141.042_RunZeroBias2023D)
- [13234.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/13234.0_TTbar_14TeV+2021FS)
- [141.044](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/141.044_RunJetMET2023D)
- [14034.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/14034.0_TTbar_14TeV+2023FS)
- [141.046](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/141.046_RunEGamma2023D)
- [14234.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/14234.0_TTbar_14TeV+2023FSPU)
- [11634.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/11634.0_TTbar_14TeV+2021)
- [2500.4](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/2500.4_NANOmc132X)
- [25202.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/25202.0_TTbar_13)
- [12434.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/12434.0_TTbar_14TeV+2023)
- [12446.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/12446.0_ZEE_14+2023)
- [12434.7](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/12434.7_TTbar_14TeV+2023_trackingMkFit)
- [10224.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/10224.0_TTbar_13+2017PU)
- [24900.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/24900.0_CloseByPGun_CE_H_Coarse_Scint+2026D98)
- [24896.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/24896.0_CloseByPGun_CE_E_Front_120um+2026D98)
- [12634.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/12634.0_TTbar_14TeV+2023PU)
- [24834.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/24834.0_TTbar_14TeV+2026D98)
- [23234.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/23234.0_TTbar_14TeV+2026D94)
- [250202.181](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/250202.181_TTbar13TeVPUppmx2018)
- [25034.999](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/25034.999_TTbar_14TeV+2026D98PU_PMXS1S2PR)
- [24834.911](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36826/runTheMatrix-results/24834.911_TTbar_14TeV+2026D98_DD4hep)
</details>

smuzaffar commented:
please test with https://github.com/cms-sw/cmsdist/pull/8936 for CMSSW_14_0_ROOT6_X

cmsbuild commented:
-1

**Failed Tests**: UnitTests 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36833/summary.html
**COMMIT**: 55145f099aed8f511450cdb0f9577e5113cd663c
**CMSSW**: CMSSW_14_0_ROOT6_X_2024-01-15-2300/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/root/193/36833/install.sh` to create a dev area with all the needed externals and cmssw changes.



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - @cms-sw cms-sw/cmsdist#8941
 - @belforte cms-sw/cmsdist#8940
 - @cms-sw cms-sw/cmsdist#8932

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36833/git-recent-commits.json
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-78f86a/36833/git-merge-result


## Unit Tests

I found 1 errors in the following unit tests:

<pre>---> test testAlcaBeamMonitor had ERRORS
</pre>

## Comparison Summary



**Summary**:
 - You potentially [added 14080 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_0_X_2024-01-15-2300+78f86a/60579/validateJR/logRootQA-events.log) to the logs
 - [Reco comparison results: 58 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_0_X_2024-01-15-2300+78f86a/60579/validateJR.html)
 - DQMHistoTests: Total files compared: 10
 - DQMHistoTests: Total histograms compared: 327017
 - DQMHistoTests: [Total failures: 225](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_0_X_2024-01-15-2300+78f86a/60579/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 326784
 - DQMHistoTests: Total skipped: 8
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 9 files compared)
 - Checked 149 log files, 71 edm output root files, 10 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_0_X_2024-01-15-2300+78f86a/60579/triggerResults): no differences found
